Sources/llbuild2fx/TaskCancellationRegistry.swift

```diff
@@ -0,0 +1,54 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+import Foundation
+import NIOConcurrencyHelpers
+
+public final class TaskCancellationRegistry {
+
+    private var triggers: [UUID: () -> Void] = [:]
```

Comment:
Rather than a raw NIOLock, should this be something like a NIOLockedValueBox around that map?

Reply:
Ah yeah, I was trying to think of what it was called. Does it matter though?

Reply:
updated to NIOLockedValueBox